be/src/olap/types.h

```diff
@@ -427,6 +459,22 @@ struct FieldTypeTraits<OLAP_FIELD_TYPE_DATE> : public BaseFieldtypeTraits<OLAP_F
         strftime(buf, sizeof(buf), "%Y-%m-%d", &time_tm);
         return std::string(buf);
     }
+    static OLAPStatus convert_from(void* dest, const void* src, const TypeInfo* src_type, MemPool* mem_pool) {
+        //only support datetime now
+        if (src_type->type() == FieldType::OLAP_FIELD_TYPE_DATETIME) {
+            using SrcType = typename CppTypeTraits<OLAP_FIELD_TYPE_DATETIME>::CppType;
+            SrcType src_value = *reinterpret_cast<const SrcType*>(src);
+            //only need part one
+            SrcType part1 = (src_value / 1000000L);
+            CppType year = static_cast<CppType>((part1 / 10000L) % 10000) - 1900;
+            CppType mon = static_cast<CppType>((part1 / 100) % 100) - 1;
+            CppType mday = static_cast<CppType>(part1 % 100);
+            CppType value = (year + 1900) * 16 * 32 + (mon + 1) * 32 + mday;
```

Comment:
```suggestion
            int value = year * 16 * 32 + mon * 32 + mday;
```

Reply:
It is better to use CppType to be compatible with date type definition